src/public/app/widgets/banner_message.js

```diff
@@ -0,0 +1,174 @@
+
+const TLP = `
+    <div id="banner-message" class="empty">
+        <style>       
+            @keyframes bannerMessageTimer {
+                0% {
+                    transform: scaleX(1);  
+                }
+                100% {
+                    transform: scaleX(0);
+                }
+            }
+            
+            @keyframes bannerMessageTimerIndefinite {
+                0%, 100% {
+                    opacity: 0.4;
+                }
+                40% {
+                    opacity: 0;
+                }
+            }
+         
+            #banner-message {
+                text-align: center;
+                font-weight: 900;
+                font-size: 1rem;
+                width: 100%;
+                color: var(--banner-color);
+                
+                position: relative;
+            }
+            
+            #banner-message.error {
+                background-color: var(--banner-background-color-error);
+            }
+            
+            #banner-message.info {
+                background-color: var(--banner-background-color-info);
+            }
+            
+            #banner-message.warning {
+                background-color: var(--banner-background-color-warning);
+            }
+            
+            #banner-message.success {
+                background-color: var(--banner-background-color-success);
+            }
+            
+            #banner-message.plain {
+                background-color: var(--banner-background-color-plain);
+            }
+            
+            #banner-message > p {
+                margin: 0;
+                padding: 5px 20px;
+            }
+            
+            #banner-message.empty > p {
+                padding: 0;
+            }
+            
+            #banner-message > .timer {
+                position: absolute;
+                left: 0;
+                bottom: 0;
+                width: 100%;
+                height: 1px;
+                background: #fff;
+                opacity: .4;
+                animation: bannerMessageTimer linear 5s forwards;
+                transform-origin: left;
+                will-change: transform;
+            }
+            
+            #banner-message > .timer.indefinite {
+                animation: bannerMessageTimerIndefinite linear 1s infinite;
+            }
+        </style>
+        <p></p>
+        <div class="timer"></div>
+    </div>
+`;
+
+const AVAILABLE_TYPES = new Set([
+    "error", "info", "warning", "success", "plain"
+]);
+
+export default class BannerMessageWidget extends BasicWidget {
+    durationTimer;
+
+    constructor() {
+        super();
+    }
+
+    doRender() {
+        this.$widget = $(TLP);
+        this.$bannerParagraph = this.$widget.find("p");
+        this.$timer = this.$widget.find(".timer");
+
+        this.$widget.on("mouseenter", this.pauseTimer.bind(this));
+        this.$widget.on("mouseleave", this.resumeTimer.bind(this));
```

Comment:
Can you clarify what's going on here?

In my mind it's really simple - (non-pausable) timer, when it detects that the network is down it will show the banner, once it's up again, banner disappears. But here I see info, warning, success, plain, what is all that for?

Reply:
In case we want to show other notifications, they should be pauseable

Reply:
To be honest I don't understand what "pausable notification" is, but in general YAGNI principle applies here.